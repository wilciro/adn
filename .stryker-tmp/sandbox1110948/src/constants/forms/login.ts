// @ts-nocheck
function stryNS_9fa48() {
  var g = new Function("return this")();
  var ns = g.__stryker__ || (g.__stryker__ = {});

  if (ns.activeMutant === undefined && g.process && g.process.env && g.process.env.__STRYKER_ACTIVE_MUTANT__) {
    ns.activeMutant = g.process.env.__STRYKER_ACTIVE_MUTANT__;
  }

  function retrieveNS() {
    return ns;
  }

  stryNS_9fa48 = retrieveNS;
  return retrieveNS();
}

stryNS_9fa48();

function stryCov_9fa48() {
  var ns = stryNS_9fa48();
  var cov = ns.mutantCoverage || (ns.mutantCoverage = {
    static: {},
    perTest: {}
  });

  function cover() {
    var c = cov.static;

    if (ns.currentTestId) {
      c = cov.perTest[ns.currentTestId] = cov.perTest[ns.currentTestId] || {};
    }

    var a = arguments;

    for (var i = 0; i < a.length; i++) {
      c[a[i]] = (c[a[i]] || 0) + 1;
    }
  }

  stryCov_9fa48 = cover;
  cover.apply(null, arguments);
}

function stryMutAct_9fa48(id) {
  var ns = stryNS_9fa48();

  function isActive(id) {
    if (ns.activeMutant === id) {
      if (ns.hitCount !== void 0 && ++ns.hitCount > ns.hitLimit) {
        throw new Error('Stryker: Hit count limit reached (' + ns.hitCount + ')');
      }

      return true;
    }

    return false;
  }

  stryMutAct_9fa48 = isActive;
  return isActive(id);
}

import { CustomFormFieldProps } from 'components/CustomForm';
export const loginForm: Array<CustomFormFieldProps> = stryMutAct_9fa48("170") ? [] : (stryCov_9fa48("170"), [stryMutAct_9fa48("171") ? {} : (stryCov_9fa48("171"), {
  label: stryMutAct_9fa48("172") ? "" : (stryCov_9fa48("172"), 'Nombre de usuario'),
  name: stryMutAct_9fa48("173") ? "" : (stryCov_9fa48("173"), 'username'),
  placeholder: stryMutAct_9fa48("174") ? "" : (stryCov_9fa48("174"), 'Ingrese el nombre de usuario'),
  type: stryMutAct_9fa48("175") ? "" : (stryCov_9fa48("175"), 'text'),
  required: stryMutAct_9fa48("176") ? {} : (stryCov_9fa48("176"), {
    value: stryMutAct_9fa48("177") ? false : (stryCov_9fa48("177"), true),
    message: stryMutAct_9fa48("178") ? "" : (stryCov_9fa48("178"), 'Por favor ingrese el nombre de usuario')
  })
}), stryMutAct_9fa48("179") ? {} : (stryCov_9fa48("179"), {
  label: stryMutAct_9fa48("180") ? "" : (stryCov_9fa48("180"), 'Contrase単a'),
  name: stryMutAct_9fa48("181") ? "" : (stryCov_9fa48("181"), 'password'),
  placeholder: stryMutAct_9fa48("182") ? "" : (stryCov_9fa48("182"), 'Ingrese la contrase単a'),
  type: stryMutAct_9fa48("183") ? "" : (stryCov_9fa48("183"), 'password'),
  required: stryMutAct_9fa48("184") ? {} : (stryCov_9fa48("184"), {
    value: stryMutAct_9fa48("185") ? false : (stryCov_9fa48("185"), true),
    message: stryMutAct_9fa48("186") ? "" : (stryCov_9fa48("186"), 'Por favor ingrese la contrase単a')
  })
})]);
export const validateFormLogin = stryMutAct_9fa48("187") ? {} : (stryCov_9fa48("187"), {
  username: stryMutAct_9fa48("188") ? () => undefined : (stryCov_9fa48("188"), (value: string) => (stryMutAct_9fa48("192") ? `${value}`.length <= 0 : stryMutAct_9fa48("191") ? `${value}`.length >= 0 : stryMutAct_9fa48("190") ? false : stryMutAct_9fa48("189") ? true : (stryCov_9fa48("189", "190", "191", "192"), (stryMutAct_9fa48("193") ? `` : (stryCov_9fa48("193"), `${value}`)).length > 0)) ? null : stryMutAct_9fa48("194") ? "" : (stryCov_9fa48("194"), 'Por favor ingrese el nombre de usuario')),
  password: stryMutAct_9fa48("195") ? () => undefined : (stryCov_9fa48("195"), (value: string) => (stryMutAct_9fa48("199") ? `${value}`.length <= 0 : stryMutAct_9fa48("198") ? `${value}`.length >= 0 : stryMutAct_9fa48("197") ? false : stryMutAct_9fa48("196") ? true : (stryCov_9fa48("196", "197", "198", "199"), (stryMutAct_9fa48("200") ? `` : (stryCov_9fa48("200"), `${value}`)).length > 0)) ? null : stryMutAct_9fa48("201") ? "" : (stryCov_9fa48("201"), 'Por favor ingrese la contrase単a'))
});
export const initialFormLogin = stryMutAct_9fa48("202") ? {} : (stryCov_9fa48("202"), {
  username: stryMutAct_9fa48("203") ? "Stryker was here!" : (stryCov_9fa48("203"), ''),
  password: stryMutAct_9fa48("204") ? "Stryker was here!" : (stryCov_9fa48("204"), '')
});