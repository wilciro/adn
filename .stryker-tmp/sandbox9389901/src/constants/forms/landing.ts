// @ts-nocheck
function stryNS_9fa48() {
  var g = new Function("return this")();
  var ns = g.__stryker__ || (g.__stryker__ = {});

  if (ns.activeMutant === undefined && g.process && g.process.env && g.process.env.__STRYKER_ACTIVE_MUTANT__) {
    ns.activeMutant = g.process.env.__STRYKER_ACTIVE_MUTANT__;
  }

  function retrieveNS() {
    return ns;
  }

  stryNS_9fa48 = retrieveNS;
  return retrieveNS();
}

stryNS_9fa48();

function stryCov_9fa48() {
  var ns = stryNS_9fa48();
  var cov = ns.mutantCoverage || (ns.mutantCoverage = {
    static: {},
    perTest: {}
  });

  function cover() {
    var c = cov.static;

    if (ns.currentTestId) {
      c = cov.perTest[ns.currentTestId] = cov.perTest[ns.currentTestId] || {};
    }

    var a = arguments;

    for (var i = 0; i < a.length; i++) {
      c[a[i]] = (c[a[i]] || 0) + 1;
    }
  }

  stryCov_9fa48 = cover;
  cover.apply(null, arguments);
}

function stryMutAct_9fa48(id) {
  var ns = stryNS_9fa48();

  function isActive(id) {
    if (ns.activeMutant === id) {
      if (ns.hitCount !== void 0 && ++ns.hitCount > ns.hitLimit) {
        throw new Error('Stryker: Hit count limit reached (' + ns.hitCount + ')');
      }

      return true;
    }

    return false;
  }

  stryMutAct_9fa48 = isActive;
  return isActive(id);
}

import { CustomFormFieldProps } from 'components/CustomForm';
export const landingForm: Array<CustomFormFieldProps> = stryMutAct_9fa48("118") ? [] : (stryCov_9fa48("118"), [stryMutAct_9fa48("119") ? {} : (stryCov_9fa48("119"), {
  label: stryMutAct_9fa48("120") ? "" : (stryCov_9fa48("120"), 'Nombre del dueño'),
  name: stryMutAct_9fa48("121") ? "" : (stryCov_9fa48("121"), 'ownerName'),
  placeholder: stryMutAct_9fa48("122") ? "" : (stryCov_9fa48("122"), 'Ingrese el nombre del dueño'),
  type: stryMutAct_9fa48("123") ? "" : (stryCov_9fa48("123"), 'text'),
  required: stryMutAct_9fa48("124") ? {} : (stryCov_9fa48("124"), {
    value: stryMutAct_9fa48("125") ? false : (stryCov_9fa48("125"), true),
    message: stryMutAct_9fa48("126") ? "" : (stryCov_9fa48("126"), 'Por favor ingrese el nombre del dueño')
  })
}), stryMutAct_9fa48("127") ? {} : (stryCov_9fa48("127"), {
  label: stryMutAct_9fa48("128") ? "" : (stryCov_9fa48("128"), 'Documento del dueño'),
  name: stryMutAct_9fa48("129") ? "" : (stryCov_9fa48("129"), 'owner_document'),
  placeholder: stryMutAct_9fa48("130") ? "" : (stryCov_9fa48("130"), 'Ingrese el documento del dueño'),
  type: stryMutAct_9fa48("131") ? "" : (stryCov_9fa48("131"), 'text'),
  required: stryMutAct_9fa48("132") ? {} : (stryCov_9fa48("132"), {
    value: stryMutAct_9fa48("133") ? false : (stryCov_9fa48("133"), true),
    message: stryMutAct_9fa48("134") ? "" : (stryCov_9fa48("134"), 'Por favor ingrese el documento del dueño')
  })
}), stryMutAct_9fa48("135") ? {} : (stryCov_9fa48("135"), {
  label: stryMutAct_9fa48("136") ? "" : (stryCov_9fa48("136"), 'Nombre de la mascota'),
  name: stryMutAct_9fa48("137") ? "" : (stryCov_9fa48("137"), 'pet_name'),
  placeholder: stryMutAct_9fa48("138") ? "" : (stryCov_9fa48("138"), 'Ingrese el nombre de la mascota'),
  type: stryMutAct_9fa48("139") ? "" : (stryCov_9fa48("139"), 'text'),
  required: stryMutAct_9fa48("140") ? {} : (stryCov_9fa48("140"), {
    value: stryMutAct_9fa48("141") ? false : (stryCov_9fa48("141"), true),
    message: stryMutAct_9fa48("142") ? "" : (stryCov_9fa48("142"), 'Por favor ingrese el nombre de la mascota')
  })
}), stryMutAct_9fa48("143") ? {} : (stryCov_9fa48("143"), {
  label: stryMutAct_9fa48("144") ? "" : (stryCov_9fa48("144"), 'Edad de la mascota'),
  name: stryMutAct_9fa48("145") ? "" : (stryCov_9fa48("145"), 'pet_age'),
  placeholder: stryMutAct_9fa48("146") ? "" : (stryCov_9fa48("146"), 'Ingrese la edad de la mascota'),
  type: stryMutAct_9fa48("147") ? "" : (stryCov_9fa48("147"), 'text'),
  required: stryMutAct_9fa48("148") ? {} : (stryCov_9fa48("148"), {
    value: stryMutAct_9fa48("149") ? false : (stryCov_9fa48("149"), true),
    message: stryMutAct_9fa48("150") ? "" : (stryCov_9fa48("150"), 'Por favor ingrese la edad de la mascota')
  })
}), stryMutAct_9fa48("151") ? {} : (stryCov_9fa48("151"), {
  label: stryMutAct_9fa48("152") ? "" : (stryCov_9fa48("152"), 'Fecha de la cita'),
  name: stryMutAct_9fa48("153") ? "" : (stryCov_9fa48("153"), 'date'),
  placeholder: stryMutAct_9fa48("154") ? "" : (stryCov_9fa48("154"), 'Seleccione la fecha de la cita'),
  type: stryMutAct_9fa48("155") ? "" : (stryCov_9fa48("155"), 'text'),
  required: stryMutAct_9fa48("156") ? {} : (stryCov_9fa48("156"), {
    value: stryMutAct_9fa48("157") ? false : (stryCov_9fa48("157"), true),
    message: stryMutAct_9fa48("158") ? "" : (stryCov_9fa48("158"), 'Por favor seleccione la fecha de la cita')
  })
})]);
export const validateFormLanding = stryMutAct_9fa48("159") ? {} : (stryCov_9fa48("159"), {
  ownerName: stryMutAct_9fa48("160") ? () => undefined : (stryCov_9fa48("160"), (value: string) => (stryMutAct_9fa48("164") ? `${value}`.length <= 0 : stryMutAct_9fa48("163") ? `${value}`.length >= 0 : stryMutAct_9fa48("162") ? false : stryMutAct_9fa48("161") ? true : (stryCov_9fa48("161", "162", "163", "164"), (stryMutAct_9fa48("165") ? `` : (stryCov_9fa48("165"), `${value}`)).length > 0)) ? null : stryMutAct_9fa48("166") ? "" : (stryCov_9fa48("166"), 'Por favor ingrese el nombre del dueño')),
  date: stryMutAct_9fa48("167") ? () => undefined : (stryCov_9fa48("167"), (value: string) => stryMutAct_9fa48("168") ? (stryMutAct_9fa48("169") ? `` : (stryCov_9fa48("169"), `${value}`)).length - 1 : (stryCov_9fa48("168"), (stryMutAct_9fa48("169") ? `` : (stryCov_9fa48("169"), `${value}`)).length + 1))
});