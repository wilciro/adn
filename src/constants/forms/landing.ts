import { CustomFormFieldProps } from 'components/CustomForm';

export const landingForm: Array<CustomFormFieldProps> = [
  {
    label: 'Nombre del dueño',
    name: 'ownerName',
    placeholder: 'Ingrese el nombre del dueño',
    type: 'text',
    required: {
      value: true,
      message: 'Por favor ingrese el nombre del dueño',
    },
  },
  {
    label: 'Documento del dueño',
    name: 'ownerDocument',
    placeholder: 'Ingrese el documento del dueño',
    type: 'text',
    required: {
      value: true,
      message: 'Por favor ingrese el documento del dueño',
    },
  },
  {
    label: 'Nombre de la mascota',
    name: 'petName',
    placeholder: 'Ingrese el nombre de la mascota',
    type: 'text',
    required: {
      value: true,
      message: 'Por favor ingrese el nombre de la mascota',
    },
  },
  {
    label: 'Edad de la mascota',
    name: 'petAge',
    placeholder: 'Ingrese la edad de la mascota',
    type: 'text',
    required: {
      value: true,
      message: 'Por favor ingrese la edad de la mascota',
    },
  },
  {
    label: 'Tipo de mascota',
    name: 'petType',
    placeholder: 'Seleccione el tipo de mascota',
    type: 'select',
    data: [
      { value: 'gato', label: 'Gato' },
      { value: 'pato', label: 'Perro' },
    ],
    required: {
      value: true,
      message: 'Por favor seleccione el tipo de mascota',
    },
  },
  {
    label: 'Fecha de la cita',
    name: 'date',
    placeholder: 'Seleccione la fecha de la cita',
    type: 'date',
    required: {
      value: true,
      message: 'Por favor seleccione la fecha de la cita',
    },
  },
  {
    label: 'Hora de la cita',
    name: 'time',
    placeholder: 'Seleccione la hora de la cita',
    type: 'time',
    required: {
      value: true,
      message: 'Por favor seleccione la hora de la cita',
    },
  },
];

export const validateFormLanding = {
  ownerName: (value: string) =>
    `${value}`.length > 0 ? null : 'Por favor ingrese el nombre del dueño',
  ownerDocument: (value: string) =>
    `${value}`.length > 0 ? null : 'Por favor ingrese el documento',
  petName: (value: string) =>
    `${value}`.length > 0 ? null : 'Por favor ingrese el nombre',
  petAge: (value: string) =>
    `${value}`.length > 0 ? null : 'Por favor ingrese la edad',
  date: (value: string) =>
    `${value}`.length > 0 ? null : 'Por favor ingrese la fecha',
  time: (value: string) =>
    `${value}`.length > 0 ? null : 'Por favor ingrese la hora',
  petType: (value: string) =>
    `${value}`.length > 0 ? null : 'Por favor seleccione el tipo',
};

export const initialFormLanding = {
  ownerName: '',
  ownerDocument: '',
  petName: '',
  petAge: '',
  date: '',
  time: new Date(),
  petType: '',
};
