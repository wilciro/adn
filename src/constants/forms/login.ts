export const loginForm = [
    {
        "label": "Username",
        "name": "username",
        "placeholder": "Type the username",
        "required": {
            "value" : true,
            "message" : "Please enter the username"
        }
    }
]