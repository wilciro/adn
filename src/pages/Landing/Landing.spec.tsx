import React from 'react';
import {
  act,
  fireEvent,
  render,
  RenderOptions,
  screen,
  waitFor,
} from '@testing-library/react';
import { landingForm } from 'constants/forms/landing';
import { NotificationsProvider } from '@mantine/notifications';
import LandingPage from '.';

describe('LandingPage tests', () => {
  /* test('should match snapshot LandingPage', () => {
    const { container } = render(<LandingPage />);

    expect(container).toMatchSnapshot();
  }); */
  test('shouldnt register', async () => {
    // const { getByText } = render(<LandingPage />);
    let container: RenderOptions<
      typeof import('@testing-library/dom/types/queries'),
      HTMLElement,
      HTMLElement
    >;
    await act(async () => {
      render(<LandingPage />, container);
    });

    await waitFor(() => {
      expect(screen.getByText('Nombre de la mascota')).toBeInTheDocument();
    });
    fireEvent.click(screen.getByTestId('register-btn'));
    expect(screen.getByText('Por favor ingrese el nombre')).toBeInTheDocument();
  });
  test('should register', async () => {
    const { container } = render(
      <NotificationsProvider>
        <LandingPage />
      </NotificationsProvider>,
    );
    fireEvent.change(screen.getByTestId(landingForm[0].name), {
      target: { value: 'Wilfer' },
    });
    fireEvent.change(screen.getByTestId(landingForm[1].name), {
      target: { value: '123' },
    });
    fireEvent.change(screen.getByTestId(landingForm[2].name), {
      target: { value: 'Ciri' },
    });
    fireEvent.change(screen.getByTestId(landingForm[3].name), {
      target: { value: 15 },
    });
    fireEvent.click(screen.getByTestId('date'));
    fireEvent.click(
      container
        .getElementsByClassName('.mantine-DatePicker-calendarHeaderControl')
        .item(0) || screen.getByTestId('date'),
    );
    fireEvent.click(
      container.getElementsByClassName('.mantine-DatePicker-day').item(20) ||
        screen.getByTestId('date'),
    );

    fireEvent.click(screen.getByTestId('time'));
    fireEvent.change(
      container.getElementsByTagName('input').item(0) ||
        screen.getByTestId('date'),
      { target: { value: 10 } },
    );

    fireEvent.click(screen.getByTestId('register-btn'));
    // expect(screen.getByText('Nombre de la mascota')).toBeTruthy();
    /* await waitFor(() => {
      expect(
        screen.getByText('Petición realizada con éxito'),
      ).toBeInTheDocument();
    }); */
  });
});
