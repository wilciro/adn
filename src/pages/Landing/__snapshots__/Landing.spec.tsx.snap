// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPage tests should match snapshot AboutPage 1`] = `
<div>
  <div
    class="mantine-yv0or0"
  >
    <h2>
      Solicitud de cita veterinaria
    </h2>
    <p>
      Diligencia todos los campos para asignar la cita
    </p>
    <div
      class="mantine-1ne3752"
    />
    <form>
      <div>
        <div
          class="mantine-TextInput-root mantine-1m3pqry"
        >
          <label
            class="mantine-1w07r5r mantine-TextInput-label"
            for="mantine-i21g8a9iu"
            id="mantine-i21g8a9iu-label"
          >
            Nombre del dueño
            <span
              aria-hidden="true"
              class="mantine-1m203yh mantine-TextInput-required"
            >
               *
            </span>
          </label>
          <div
            class="mantine-TextInput-wrapper mantine-12sbrde"
          >
            <input
              aria-invalid="false"
              class="mantine-TextInput-defaultVariant mantine-TextInput-input mantine-g773hu"
              data-testid="Nombre del dueño"
              id="mantine-i21g8a9iu"
              placeholder="Ingrese el nombre del dueño"
              required=""
              type="text"
            />
          </div>
        </div>
      </div>
      <div>
        <div
          class="mantine-TextInput-root mantine-1m3pqry"
        >
          <label
            class="mantine-1w07r5r mantine-TextInput-label"
            for="mantine-3v26sm8sn"
            id="mantine-3v26sm8sn-label"
          >
            Documento del dueño
            <span
              aria-hidden="true"
              class="mantine-1m203yh mantine-TextInput-required"
            >
               *
            </span>
          </label>
          <div
            class="mantine-TextInput-wrapper mantine-12sbrde"
          >
            <input
              aria-invalid="false"
              class="mantine-TextInput-defaultVariant mantine-TextInput-input mantine-g773hu"
              data-testid="Documento del dueño"
              id="mantine-3v26sm8sn"
              placeholder="Ingrese el documento del dueño"
              required=""
              type="text"
            />
          </div>
        </div>
      </div>
      <div>
        <div
          class="mantine-TextInput-root mantine-1m3pqry"
        >
          <label
            class="mantine-1w07r5r mantine-TextInput-label"
            for="mantine-i68zcsq0v"
            id="mantine-i68zcsq0v-label"
          >
            Nombre de la mascota
            <span
              aria-hidden="true"
              class="mantine-1m203yh mantine-TextInput-required"
            >
               *
            </span>
          </label>
          <div
            class="mantine-TextInput-wrapper mantine-12sbrde"
          >
            <input
              aria-invalid="false"
              class="mantine-TextInput-defaultVariant mantine-TextInput-input mantine-g773hu"
              data-testid="Nombre de la mascota"
              id="mantine-i68zcsq0v"
              placeholder="Ingrese el nombre de la mascota"
              required=""
              type="text"
            />
          </div>
        </div>
      </div>
      <div>
        <div
          class="mantine-TextInput-root mantine-1m3pqry"
        >
          <label
            class="mantine-1w07r5r mantine-TextInput-label"
            for="mantine-jt9qmimot"
            id="mantine-jt9qmimot-label"
          >
            Edad de la mascota
            <span
              aria-hidden="true"
              class="mantine-1m203yh mantine-TextInput-required"
            >
               *
            </span>
          </label>
          <div
            class="mantine-TextInput-wrapper mantine-12sbrde"
          >
            <input
              aria-invalid="false"
              class="mantine-TextInput-defaultVariant mantine-TextInput-input mantine-g773hu"
              data-testid="Edad de la mascota"
              id="mantine-jt9qmimot"
              placeholder="Ingrese la edad de la mascota"
              required=""
              type="text"
            />
          </div>
        </div>
      </div>
      <div>
        <div
          class="mantine-TextInput-root mantine-1m3pqry"
        >
          <label
            class="mantine-1w07r5r mantine-TextInput-label"
            for="mantine-9mjytqx6s"
            id="mantine-9mjytqx6s-label"
          >
            Fecha de la cita
            <span
              aria-hidden="true"
              class="mantine-1m203yh mantine-TextInput-required"
            >
               *
            </span>
          </label>
          <div
            class="mantine-TextInput-wrapper mantine-12sbrde"
          >
            <input
              aria-invalid="false"
              class="mantine-TextInput-defaultVariant mantine-TextInput-input mantine-g773hu"
              data-testid="Fecha de la cita"
              id="mantine-9mjytqx6s"
              placeholder="Seleccione la fecha de la cita"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </form>
    <div
      class="mantine-1ne3752"
    />
    <button
      class="mantine-Button-filled mantine-Button-root mantine-1grzg0q"
      data-testid="register-btn"
      type="button"
    >
      <div
        class="mantine-3xbgk5 mantine-Button-inner"
      >
        <span
          class="mantine-qo1k2 mantine-Button-label"
        >
          Registrar
        </span>
      </div>
    </button>
  </div>
</div>
`;
